server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1747,7 +1747,7 @@ public VMTemplateVO createPrivateTemplateRecord(CreateTemplateCmd cmd, Account t
         _accountMgr.checkAccess(caller, null, true, templateOwner);
 
         String name = cmd.getTemplateName();
-        if ((name == null) || (name.length() > 32)) {
+        if ((name == null) || (name.length() > 255)) {
             throw new InvalidParameterValueException("Template name cannot be null and should be less than 32 characters");
```

Comment:
Fix the error msg to say 255 characters too - otherwise LGTM (also check any limitation in the db column, VO etc)

Reply:
```suggestion
            throw new InvalidParameterValueException("Template name cannot be null and should be less than 255 characters");
```